src/lib/integration_test_cloud_engine/mina_automation.ml

```diff
@@ -812,7 +812,7 @@ module Network_manager = struct
     let%bind () = File_system.remove_dir t.testnet_dir in
     Deferred.unit
 
-  let generate_code_coverage t network =
+  let tear_down t network =
```

Comment:
Why doesn't this just happen at the beginning of `cleanup`? Do we need an additional function for this step?

Reply:
This is something we can discuss further. Main question is how we want to react on possible failure of coverage gathering. If we move this code to clean up (which is not returning errors) then we can only silence it or panic. In current version tear_down return Malleable_error, so handling error is possible in future code. I'm not sure which approach is better. Do you have some opinion ?